src/ucp/api/ucp.cuh

```diff
@@ -0,0 +1,264 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2025. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCP_CUH
+#define UCP_CUH
+
+#include <ucs/type/status.h>
+#include <uct/api/uct.h>
+#include <ucp/api/ucp.h>
+
+
+/* TODO: Use from ucp.h */
+typedef struct ucp_dlist_elem {} ucp_dlist_elem_t;
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief GPU request descriptor of a given batch
+ *
+ * This request tracks a batch of memory operations in progress. It can be used
+ * with @ref ucp_gpu_progress_req to detect request completion.
+ */
+typedef struct ucp_gpu_request {
+} ucp_gpu_request_t;
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Descriptor list handle stored on GPU memory.
+ *
+ * This handle is obtained and managed with functions called on host. It can be
+ * used repeatedly from GPU code to perform memory transfers.
+ *
+ * The handle and most of its content is stored on GPU memory, with the intent
+ * to be as memory-local as possible.
+ */
+typedef struct {
```

Comment:
this structure, and also ucp_dlist_elem_t, cannot be defined neither in ucp.cuh nor in ucp.h.
not in ucp.cuh because they need to be included from host export code.
not from ucp.h because they cannot be exposed in host api.
so need to define them in src/ucp/apu/cuda/ucp_cuda_types.h or similar.